tiny_cnn/layers/convolutional_layer.h

```diff
@@ -415,108 +343,125 @@ class convolutional_layer : public feedforward_layer<Activation> {
                          bool           has_bias,
                          cnn_size_t     w_stride,
                          cnn_size_t     h_stride) {
-        in_ = in;
-        in_padded_ = shape3d(in_length(in.width_, w_width, ptype),
-                             in_length(in.height_, w_height, ptype),
-                             in.depth_);
-        out_ = shape3d(conv_out_length(in.width_, w_width, w_stride, ptype),
-                       conv_out_length(in.height_, w_height, h_stride, ptype),
-                       outc);
-        weight_ = shape3d(w_width, w_height, in.depth_ * outc);
-        has_bias_ = has_bias;
-        pad_type_ = ptype;
-        w_stride_ = w_stride;
-        h_stride_ = h_stride;
+        params_.in = in;
+        params_.in_padded = shape3d(in_length(in.width_, w_width, ptype),
+                                    in_length(in.height_, w_height, ptype),
+                                    in.depth_);
+        params_.out =
+            shape3d(conv_out_length(in.width_, w_width, w_stride, ptype),
+                    conv_out_length(in.height_, w_height, h_stride, ptype),
+                    outc);
+        params_.weight = shape3d(w_width, w_height, in.depth_ * outc);
+        params_.has_bias = has_bias;
+        params_.pad_type = ptype;
+        params_.w_stride = w_stride;
+        params_.h_stride = h_stride;
     }
 
     void init() {
-        for (conv_layer_worker_specific_storage& cws : conv_layer_worker_storage_) {
-            if (pad_type_ == padding::same) {
-                cws.prev_out_buf_.resize(in_padded_.size(), float_t(0));
-                cws.prev_delta_padded_.resize(in_padded_.size(), float_t(0));
-            }
-            else {
+        // TODO: got a seffault at somepoint here
+        for (conv_layer_worker_specific_storage& cws :
+                conv_layer_worker_storage_) {
+            if (params_.pad_type == padding::same) {
+                cws.prev_out_buf_.resize(params_.in_padded.size(), float_t(0));
+                cws.prev_delta_padded_.resize(
+                    params_.in_padded.size(), float_t(0));
+            } else {
                 cws.prev_out_buf_.clear();
```

Comment:
@nyanp something weird is happening here when we allocate workers and we pass them to kernels. A segfault raises in this case.